src/debugger/Debugger.cpp

```diff
@@ -246,6 +264,82 @@ bool Debugger::processIncomingMessages(ExecutionState* state)
     return enabled();
 }
 
+bool Debugger::doEval(ExecutionState* state, ByteCodeBlock* byteCodeBlock, uint8_t* buffer, size_t length)
+{
+    uint8_t type = (uint8_t)(buffer[0] + 1);
+    uint32_t size;
+    char* data;
+
+    memcpy(&size, buffer + 1, sizeof(uint32_t));
+
+    if (size > ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH - 1 - sizeof(uint32_t)) {
+        if (length != ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH) {
+            goto error;
+        }
+
+        data = (char*)GC_MALLOC_ATOMIC(size);
+        char* ptr = data;
+        char* end = data + size;
+
+        length = ESCARGOT_DEBUGGER_MAX_MESSAGE_LENGTH - 1 - sizeof(uint32_t);
+        memcpy(ptr, buffer + 1 + sizeof(uint32_t), length);
+        ptr += length;
+
+        do {
+            if (receive(buffer, length)) {
+                if (buffer[0] != type || (length > (size_t)(end - ptr + 1))) {
+                    goto error;
+                }
+
+                memcpy(ptr, buffer + 1, length - 1);
+                ptr += length - 1;
+            } else if (!enabled()) {
+                return false;
+            }
+        } while (ptr < end);
+    } else {
+        if (size + 1 + sizeof(uint32_t) != length) {
+            goto error;
+        }
+
+        data = (char*)buffer + 1 + sizeof(uint32_t);
+    }
+
+    String* str;
+
+    if (type == ESCARGOT_MESSAGE_EVAL_8BIT) {
+        str = new Latin1String(data, size);
+    } else {
+        str = new UTF16String((char16_t*)data, size / 2);
+    }
+
+    type = ESCARGOT_MESSAGE_EVAL_RESULT_8BIT;
+    m_stopState = ESCARGOT_DEBUGGER_IN_EVAL_MODE;
+
+    try {
+        Value asValue(str);
+        Value result(state->context()->globalObject()->evalLocal(*state, asValue, Value(Value::Undefined), byteCodeBlock->m_codeBlock->asInterpretedCodeBlock(), true));
```

Comment:
I passed true for `inWithOperation`. Is this ok, or shall I pass the from the byte code interpreter?

Reply:
IMO we can just pass true. passing true just add a few step when calling function.
the step checks we should be affect by with statement. thus passing true is ok